jaxrs/src/test/java/org/jboss/as/jaxrs/JaxrsSubsystem20TestCase.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2019, Red Hat Middleware LLC, and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.jaxrs;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.RunningMode;
+import org.jboss.as.model.test.FailedOperationTransformationConfig;
+import org.jboss.as.model.test.ModelTestControllerVersion;
+import org.jboss.as.model.test.ModelTestUtils;
+import org.jboss.as.subsystem.test.AbstractSubsystemBaseTest;
+import org.jboss.as.subsystem.test.AdditionalInitialization;
+import org.jboss.as.subsystem.test.KernelServices;
+import org.jboss.as.subsystem.test.KernelServicesBuilder;
+import org.jboss.dmr.ModelNode;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * @author <a href="kabir.khan@jboss.com">Kabir Khan</a>
+ * @author <a href="mailto:ema@rehdat.com>Jim Ma</a>
+ * @author <a href="mailto:alessio.soldano@jboss.com>Alessio Soldano</a>
+ * @author <a href="rsigal@redhat.com>Ron Sigal</a>
+ */
+public class JaxrsSubsystem20TestCase extends AbstractSubsystemBaseTest {
+
+    public JaxrsSubsystem20TestCase() {
+        super(JaxrsExtension.SUBSYSTEM_NAME, new JaxrsExtension());
+    }
+
+    @Override
+    protected String getSubsystemXml() throws IOException {
+        return readResource("jaxrs-2.0.xml");
+    }
+
+    @Override
+    protected String getSubsystemXsdPath() throws Exception {
+        return "schema/jboss-as-jaxrs_2_0.xsd";
+    }
+
+    @Override
+    protected String[] getSubsystemTemplatePaths() throws IOException {
+        return new String[]{"/subsystem-templates/jaxrs.xml"};
+    }
+
+    @Override
+    public void testSubsystem() throws Exception {
+        standardSubsystemTest(null);
+    }
+
+    @Test
+    public void testTransformers() throws Exception {
+        testTransformers_2_0_to_1_0(ModelTestControllerVersion.EAP_7_2_0);
```

Comment:
We generally (although it doesn't happen *everywhere*!) use two tests for these, e.g:

* testTransformers720()
* testRejectingTransformers720()

Also, it is typically better to pull these out into a new JaxrsTransformersTestCase class since we need parser test for each model version (see my comment about deleting JaxrsSubsystemTestCase.java)

Reply:
@kabir, I've broken out JaxrsTransformersTestCase. When you talk about testTransformers720() and testRejectingTransformers720(), are you just talking about renaming the tests? That's what I've done, in fact.